polynote-server/src/main/scala/polynote/server/StreamingHandleManager.scala

```diff
@@ -0,0 +1,53 @@
+package polynote.server
+
+import java.nio.ByteBuffer
+import java.util.concurrent.ConcurrentHashMap
+
+import cats.effect.IO
+import polynote.messages.ByteVector32
+import polynote.runtime.StreamingDataRepr
+import scodec.bits.ByteVector
+
+import scala.collection.mutable
+
+/**
+  * Manages multiplexed state of streaming data - each instance of StreamingHandleManager has its own iterator per handle
+  * at any given time.
+  */
+class StreamingHandleManager {
+  private val streams = new ConcurrentHashMap[Int, Iterator[ByteVector32]]
```

Comment:
[minor] would it be slightly more correct to have this be `HandleIterator` rather than `Iterator[ByteVector32]`? Not like we are worried about other `Iterators` sneaking in though...

Reply:
Sure, why not? I'll change it

Reply:
Fixed